h3/src/config.rs

```diff
@@ -58,14 +170,25 @@ impl Config {
 impl Default for Config {
     fn default() -> Self {
         Self {
-            max_field_section_size: VarInt::MAX.0,
             send_grease: true,
             #[cfg(test)]
             send_settings: true,
-            enable_webtransport: false,
-            enable_extended_connect: false,
-            enable_datagram: false,
-            max_webtransport_sessions: 0,
+            settings: Default::default(),
         }
     }
 }
+
+impl Deref for Config {
```

Comment:
> Perhaps one thing to discuss is whether or not to include the Deref and DerefMut impls here: https://github.com/hyperium/h3/pull/208/files#diff-4fc440b791f25fbde8061757f9ecfdcb61af28bc20a5b3c236f0ae1f6c96db4eR181-R194
> https://doc.rust-lang.org/std/ops/trait.Deref.html says that: "Deref should only be implemented for smart pointers"

`Config` is not a Smart Pointer so I think we should remove them to avoid confusion.